framework/src/utils/MooseUtils.C

```diff
@@ -575,6 +575,34 @@ wildCardMatch(std::string name, std::string search_string)
     return false;
 }
 
+template <>
+int
+convert<int>(const std::string & str)
```

Comment:
So only signed int? What about unsigned int, long, unsigned long, etc? Perhaps we should make a separate template method (not specialized version of convert) to handle all of the integer types and only call the original method for things that we probably really shouldn't allow scientific notation for.

Reply:
I agree with Cody - we should make sure we handle all the integer types.

Reply:
I didn't need those...

Reply:
Yeah-huh